src/templates/ControlsList.jsx

```diff
@@ -0,0 +1,214 @@
+      <td>
+        {control.status === "completed"
+          ? "Completed"
+          : control.status === "incomplete"
+          ? "Incomplete"
+          : "Not started"}
+      </td>
+    </>
+  );
+};
+
+const EmptyResults = ({ length }) => {
+  if (length !== 0) {
+    return "";
+  }
+  return (
+    <p>
+      There are no controls in your project yet. Add one by visiting the{" "}
+      <Link to={MAIN_ROUTES.COMPONENT_LIBRARY}>Component Library</Link>.
+    </p>
+  );
+};
+
+const FilterCol = ({ currentStatus, checkBoxHandler }) => {
+  const isCheckedStatus = (value) => {
+    if (currentStatus.includes(value)) {
+      return true;
+    }
+    return false;
+  };
+  return (
+    <>
+      <div className="component-library-filter grid-col-2">
+        <fieldset className="usa-fieldset">
+          <legend className="usa-legend">Status</legend>
+          <Checkbox
+            id="completed-filter"
+            name="status__in"
+            label="Completed"
+            value="completed"
+            onChange={checkBoxHandler}
+            key="3"
+            checked={isCheckedStatus("completed")}
+          />
+          <Checkbox
+            id="incomplete-filter"
+            name="status__in"
+            label="Incomplete"
+            value="incomplete"
+            onChange={checkBoxHandler}
+            key="2"
+            checked={isCheckedStatus("incomplete")}
+          />
+          <Checkbox
+            id="notstarted-filter"
+            name="status__in"
+            label="Not Started"
+            value="not_started"
+            onChange={checkBoxHandler}
+            key="1"
+            checked={isCheckedStatus("not_started")}
+          />
+        </fieldset>
+      </div>
+      <div className="grid-col-1"></div>
+    </>
+  );
+};
+// this seems overly complex when it is not doing much :/ defining variables, setting url params, callbacks, template varibles finished, template
+const ProjectControls = ({ controlsList = [], totalItemCount = 0 }) => {
+  const baseUrl = useLocation();
+  let navigate = useNavigate();
+  const [currentStatus, setCurrentStatus] = useState("");
+  const [currentPage, setCurrentPage] = useState(1);
+  const [queryString, setQueryString] = useState("");
+
+  //set the url query params into state
+  const [searchParams] = useSearchParams();
+  if (
+    Number(searchParams.get("page")) &&
+    currentPage !== Number(searchParams.get("page"))
+  ) {
+    setCurrentPage(Number(searchParams.get("page")));
+  }
+
+  if (
+    searchParams.get("status__in") &&
+    !currentStatus.includes(searchParams.get("status__in"))
+  ) {
+    setCurrentStatus(searchParams.get("status__in"));
+    setQueryString("status__in=" + searchParams.get("status__in"));
+  }
+
+  // Call back functions used for updating values from the page
+  //callback function to change the current page
+  const onPageChange = (pageNumber) => {
+    setCurrentPage(pageNumber);
+  };
```

Comment:
The paginated response that you implemented with rest framework should include links to next and previous pages so that you'd only actually have to construct the url once. Would that help simplify this?

Reply:
We have a ticket open for improving overall pagination construction. For now, this is following the pattern that was previously implemented.